functorch/_src/aot_autograd.py

```diff
@@ -908,56 +952,51 @@ def functional_call(*args, **kwargs):
             )
         return out
 
-    def aot_function_simplified(
-        fn: Callable,
-        fw_compiler: Callable,
-        bw_compiler: Optional[Callable] = None,
-        partition_fn: Callable = default_partition,
-        decompositions: Optional[Dict] = None,
-        hasher_type=None,
-        static_argnums=None,
-    ) -> Callable:
-        assert static_argnums is None
-        if bw_compiler is None:
-            bw_compiler = fw_compiler
-        aot_config = AOTConfig(
-            fw_compiler=fw_compiler,
-            bw_compiler=bw_compiler,
-            partition_fn=partition_fn,
-            decompositions=decompositions,
-            num_params_buffers=params_len,
-        )
-
-        compiled_fn = None
+    assert static_argnums is None
+    if bw_compiler is None:
+        bw_compiler = fw_compiler
 
-        @wraps(fn)
-        def new_func(*args):
-            nonlocal compiled_fn
-            if compiled_fn is None:
-                compiled_fn = create_aot_dispatcher_function(
-                    fn,
-                    args,
-                    aot_config,
-                )
-            return compiled_fn(args)
+    aot_config = AOTConfig(
+        fw_compiler=fw_compiler,
+        bw_compiler=bw_compiler,
+        partition_fn=partition_fn,
+        decompositions=decompositions,
+        num_params_buffers=params_len,
+    )
 
-        return new_func
+    joined_args = fake_flat_tensor_args + list(inputs)
+    aot_dispatcher_function = _create_aot_dispatcher_function(functional_call, joined_args, aot_config, fake_mode)
 
-    compiled_f = aot_function_simplified(functional_call, *top_args, **top_kwargs)
+    @wraps(functional_call)
+    def compiled_function(*args):
+        return aot_dispatcher_function(args)
+    # [Real vs Fake Params]
+    #
+    # We have a few options of what we need to do here, but a few rules for why this is the way it is:
+    #     - The "runtime" fwd must use real params, as these get invoked alongside real args
+    #     - _create_aot_dispatcher_function must be called with fake params and fake args
+    #
+    # So, we have a few ways of handling it
+    # 1) Dynamo passes only real params in, we fakify at the aot level (this current state)
+    # 2) Dynamo passes only fake params in, but we rewrite the def forward(*args): to def forward(*parms, *args):
+    #    (Not bad, but very annoying w/r/t changing dynamo's calling convention and contracts. Also, not having to close
+    #    over params is nice, and better w/r/t lifecycles and mutations)
+    # 3) Dynamo passes both real and fake tensor in (Not a bad future, but really a stopgap to 2)
+    #
+    # (2) is the correct long term direction, and we are at (1) for now, (3) can be done to make things a little more streamlined
+    # w/r/t where fake tensor conversion happens.
```

Comment:
This note doesn't really explain why you would prefer fake or real in the first place. Clearly runtime needs real. But compile time sometimes needs real too, and it's worth spelling that out explicitly here. From the description here it's not clear what `_create_aot_dispatcher_function` means